catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/activity/rtl/RtlLayoutsTest.java

```diff
@@ -0,0 +1,247 @@
+				.perform(click());
+		onView(withId(Resources.getSystem().getIdentifier("action_mode_close_button", "id", "android")))
+				.perform(click());
+		onView(withId(android.R.id.button1))
+				.perform(click());
+		SystemClock.sleep(3000);
+
+		openActionBarOverflowOrOptionsMenu(baseActivityTestRule.getActivity());
+		assertTrue(checkTextDirection(getResourcesString(R.string.last_used)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.size)));
+		onView(withText(R.string.show_details))
+				.perform(click());
+		onView(withText(R.string.last_used))
+				.check(matches(isDisplayed()));
+		onView(withText(R.string.size))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.details_right_top))
+				.check(matches(isDisplayed()));
+		onView(withId(R.id.details_right_bottom))
+				.check(matches(isDisplayed()));
+		onView(withIndex(withId(R.id.list_item_background), 0))
+				.perform(click());
+
+		openActionBarOverflowOrOptionsMenu(baseActivityTestRule.getActivity());
+		onView(withText(R.string.show_details))
+				.perform(click());
+		onView(withIndex(withId(R.id.spritelist_background_headline), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.project_activity_sprite_title), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.spritelist_item_background), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.textView_number_of_scripts), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.textView_number_of_looks), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.textView_number_of_bricks), 0))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withIndex(withId(R.id.textView_number_of_sounds), 0))
+				.check(matches(isCompletelyDisplayed()));
+
+		assertTrue(checkTextDirection(getResourcesString(R.string.spritelist_background_headline)));
+		assertTrue(checkTextDirection(getResourcesString(R.string.sprites)));
+
+		onView(withIndex(withId(R.id.textView_number_of_scripts), 0))
+				.check(isLeftOf(withIndex(withId(R.id.sprite_img), 0)));
+		onView(withIndex(withId(R.id.textView_number_of_looks), 0))
+				.check(isLeftOf(withIndex(withId(R.id.sprite_img), 0)));
+		onView(withIndex(withId(R.id.textView_number_of_bricks), 0))
+				.check(isLeftOf(withIndex(withId(R.id.textView_number_of_scripts), 0)));
+		onView(withIndex(withId(R.id.textView_number_of_sounds), 0))
+				.check(isLeftOf(withIndex(withId(R.id.textView_number_of_looks), 0)));
+		onView(withIndex(withId(R.id.spritelist_item_background), 0))
+				.perform(click());
+		onView(withText(R.string.scripts))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withText(R.string.backgrounds))
+				.check(matches(isCompletelyDisplayed()));
+		onView(withText(R.string.sounds))
+				.check(matches(isCompletelyDisplayed()));
+
+		assertTrue(checkTextDirection(getResourcesString(R.string.scripts)));
```

Comment:
is there any specific reasons that those asserts are here in a UI test?
and does it make sense in the first place to check if every string is translated?

Reply:
> is there any specific reasons that those asserts are here in a UI test?

those asserts are replaced with a custom matcher

> and does it make sense in the first place to check if every string is translated?

could you be specific with your Question?